### DISCUSSION TITLE
No extract grains boundary

### DISCUSSION THREAD

**Discussion by JoRBN-H2** [mtex-toolbox/mtex#815] [Ask Anything] [answered]
Hi community MTEX,

I have a ebsd map of the Co-Cr alloy containing two phases (cobalt FCC (Co) and cobalt HCP (Coh)) and not achieve to extract the grain boundaries according to the phase.
Using the function:
`gB = grains.boundary; % extract all grain boundaries` 
It works perfectly.
But not using:
```
gB = grains.boundary ('Co', 'Co'); % or
gB = grains.boundary (' Coh', 'Coh'); % or
gB = grains.boundary ( 'Co', 'Coh');
```
Extract the same grain boundaries segments independently of the chosen phases.
However, on other maps this does not happen.

Someone could help me?

Best regards,
João Rodrigues

**Comment by ralfHielscher** [comment, marked as answer]
You need to provide distinguishable names, e.g.
```matlab
grains.boundary ( 'cobalt FCC', 'cobalt HCP');
```
Ralf.

**Reply by JoRBN-H2** [reply]
Hi Ralf,

Thanks for answering

This part of the code:
```
CS = {'notIndexed', crystalSymmetry('6/mmm', [2.5 2.5 4.1], 'X||a*', 'Y||b', 'Z||c', 'mineral', 'Coh', 'color', [0.53 0.81 0.98]),...
  crystalSymmetry('m-3m', [3.6 3.6 3.6], 'mineral', 'Co', 'color', [0.53 0.81 0.98])};

>> ebsd = EBSD.load(fname,CS,'interface','crc','convertEuler2SpatialReferenceFrame');

>> [grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'angle',2*degree);

>> gB = grains.boundary;
 gB = grainBoundary (show methods, plot)
Segments   mineral 1  mineral 2
       18  notIndexed        Coh
     1632  notIndexed         Co
        8         Coh        Coh
     1833         Coh         Co
     5567          Co         Co

>>gB = grains.boundary('Co','Co');
 gB = grainBoundary (show methods, plot)
  Segments  mineral 1  mineral 2
        8        Coh        Coh

>> gB = grains.boundary('Coh','Coh');
 gB = grainBoundary (show methods, plot)
  Segments  mineral 1  mineral 2
        8        Coh        Coh

>> gB = grains.boundary('Coh','Co');
 gB = grainBoundary (show methods, plot)
  Segments  mineral 1  mineral 2
        8        Coh        Coh
```
Always extract the same boundaries segments no matter of the phase transistion.

Best regards.

**Reply by ralfHielscher** [reply]
Ok, I see. The problem are your mineral names. They should not be substrings of each other.
Just replace the mineral names by something more distinguishable.

Ralf.

**Reply by JoRBN-H2** [reply]
Hi Ralf,

Thanks for help me.
It worked really well.

João Rodrigues
